src/Controller/ProductEnqueueController.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Webgriffe\SyliusAkeneoPlugin\Controller;
+
+use Sylius\Component\Product\Model\ProductInterface;
+use Sylius\Component\Product\Repository\ProductRepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Webgriffe\SyliusAkeneoPlugin\Entity\QueueItem;
+use Webgriffe\SyliusAkeneoPlugin\Repository\QueueItemRepositoryInterface;
+
+final class ProductEnqueueController extends AbstractController
+{
+    /** @var QueueItemRepositoryInterface */
+    private $queueItemRepository;
+
+    /** @var UrlGeneratorInterface */
+    private $urlGenerator;
+
+    /** @var ProductRepositoryInterface */
+    private $productRepository;
+
+    /**
+     * ProductEnqueueController constructor.
+     */
+    public function __construct(QueueItemRepositoryInterface $queueItemRepository, ProductRepositoryInterface $productRepository, UrlGeneratorInterface $urlGenerator)
+    {
+        $this->queueItemRepository = $queueItemRepository;
+        $this->urlGenerator = $urlGenerator;
+        $this->productRepository = $productRepository;
+    }
+
+    public function enqueue(int $productId): Response
```

Comment:
I'd prefer `enqueueAction` in method name. It's a de-facto standard in Sylius to use the `Action` suffix for the controller methods names.